### PR TITLE
Apply Stylint to the codebase

### PR DESCRIPTION
**Pull request by orta**
screen shot 2017-07-13 at 17 06 06

Currently just gives us inline warnings. All auto-fixing tools aren't quite there yet, so I've not added them, apart from recommending the extension. 

WIP because it still fails, due to prettier cropping template strings into a single line occasionally and stylinbt doesn't like that.

Skip new tests

### CODE REVIEW

**Comment by ArtsyOpenSource** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Fails</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:no_entry_sign:</td>
      <td>

  No CHANGELOG added.
  </td>
    </tr>
  
<tr>
      <td>:no_entry_sign:</td>
      <td>

  Missing Test Files:

- `src/lib/Components/Storybooks/__tests__/Styles-tests.tsx`

If these files are supposed to not exist, please update your PR body to include "Skip New Tests".
  </td>
    </tr>
  </tbody>
</table>


<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>

  New dependencies added: stylelint, stylelint-config-standard and stylelint-processor-styled-components.
  </td>
    </tr>
  </tbody>
</table>



<h2><a href="https://stylelint.io">stylelint</a></h2>
<p>Author: Unknown</p>
<p>Description: A mighty, modern CSS linter.</p>
<p>Homepage: <a href="https://stylelint.io">https://stylelint.io</a></p>

<table>
  <thead><tr><th></th><th width="100%"></th></tr></thead>
  <tr><td>Created</td><td>over 2 years ago</td></tr><tr><td>Last Updated</td><td>about 23 hours ago</td></tr><tr><td>License</td><td>MIT</td></tr><tr><td>Maintainers</td><td>5</td></tr><tr><td>Releases</td><td>111</td></tr><tr><td>Direct Dependencies</td><td><a href='http: //npmjs.com/package/autoprefixer'>autoprefixer</a>, <a href='http: //npmjs.com/package/balanced-match'>balanced-match</a>, <a href='http: //npmjs.com/package/chalk'>chalk</a>, <a href='http: //npmjs.com/package/colorguard'>colorguard</a>, <a href='http: //npmjs.com/package/cosmiconfig'>cosmiconfig</a>, <a href='http: //npmjs.com/package/debug'>debug</a>, <a href='http: //npmjs.com/package/doiuse'>doiuse</a>, <a href='http: //npmjs.com/package/execall'>execall</a>, <a href='http: //npmjs.com/package/file-entry-cache'>file-entry-cache</a>, <a href='http: //npmjs.com/package/get-stdin'>get-stdin</a>, <a href='http: //npmjs.com/package/globby'>globby</a>, <a href='http: //npmjs.com/package/globjoin'>globjoin</a>, <a href='http: //npmjs.com/package/html-tags'>html-tags</a>, <a href='http: //npmjs.com/package/ignore'>ignore</a>, <a href='http: //npmjs.com/package/imurmurhash'>imurmurhash</a>, <a href='http: //npmjs.com/package/known-css-properties'>known-css-properties</a>, <a href='http: //npmjs.com/package/lodash'>lodash</a>, <a href='http: //npmjs.com/package/log-symbols'>log-symbols</a>, <a href='http: //npmjs.com/package/mathml-tag-names'>mathml-tag-names</a>, <a href='http: //npmjs.com/package/meow'>meow</a>, <a href='http: //npmjs.com/package/micromatch'>micromatch</a>, <a href='http: //npmjs.com/package/normalize-selector'>normalize-selector</a>, <a href='http: //npmjs.com/package/pify'>pify</a>, <a href='http: //npmjs.com/package/postcss'>postcss</a>, <a href='http: //npmjs.com/package/postcss-less'>postcss-less</a>, <a href='http: //npmjs.com/package/postcss-media-query-parser'>postcss-media-query-parser</a>, <a href='http: //npmjs.com/package/postcss-reporter'>postcss-reporter</a>, <a href='http: //npmjs.com/package/postcss-resolve-nested-selector'>postcss-resolve-nested-selector</a>, <a href='http: //npmjs.com/package/postcss-scss'>postcss-scss</a>, <a href='http: //npmjs.com/package/postcss-selector-parser'>postcss-selector-parser</a>, <a href='http: //npmjs.com/package/postcss-value-parser'>postcss-value-parser</a>, <a href='http: //npmjs.com/package/resolve-from'>resolve-from</a>, <a href='http: //npmjs.com/package/specificity'>specificity</a>, <a href='http: //npmjs.com/package/string-width'>string-width</a>, <a href='http: //npmjs.com/package/style-search'>style-search</a>, <a href='http: //npmjs.com/package/stylehacks'>stylehacks</a>, <a href='http: //npmjs.com/package/sugarss'>sugarss</a>, <a href='http: //npmjs.com/package/svg-tags'>svg-tags</a> and <a href='http: //npmjs.com/package/table'>table</a></td></tr><tr><td>Keywords</td><td>css, less, scss, sugarss, lint, linter and stylelint</td></tr>
</table>

<details>
<summary><code>README</code></summary>
# stylelint

[NPM version](https://www.npmjs.org/package/stylelint) [Build Status](https://travis-ci.org/stylelint/stylelint) [Build status](https://ci.appveyor.com/project/stylelint/stylelint/branch/master) [NPM Downloads](https://www.npmjs.org/package/stylelint) [Bountysource](https://www.bountysource.com/trackers/9282518-stylelint?utm_source=9282518&utm_medium=shield&utm_campaign=TRACKER_BADGE)

A mighty, modern CSS linter that helps you enforce consistent conventions and avoid errors in your stylesheets.

## Features

-   **Over one hundred and fifty rules:** Including those that:
    -   **Catch errors**: e.g. invalid hex colors, indistinguishable colors, or overriding shorthand properties.
    -   **Enforce best practices**: e.g. keeping specificity low or disallowing vendor prefixes in your source code.
    -   **Control what languages features can be used**: e.g. whitelisting specific units, properties and functions, or disallowing certain selector types.
    -   **Enforce code style conventions**: e.g. checking the spacing around the colon in declarations or specifying patterns for class selectors.
-   **Support for the latest CSS syntax:** Including custom properties, range context for media features, calc() and nesting.
-   **Understands *CSS-like* syntaxes:** The linter is powered by [PostCSS](https://github.com/postcss/postcss), so it understands any syntax that PostCSS can parse, including SCSS, [SugarSS](https://github.com/postcss/sugarss), and *experimental support* for Less.
-   **Completely unopinionated:** Only enable the rules you want, and configure them with options that tailor the linter to your needs.
-   **Support for plugins:** It's easy to create your own rules and add them to the linter.
-   **Automatically fixes some stylistic warnings:** Save time by having stylelint fix your code with this *experimental* feature.
-   **Shareable configs:** If you don't want to craft your own config, you can extend a shareable config.
-   **Options validator:** So that you can be confident that your config is valid.
-   **Well tested:** Nearly twenty five thousand tests cover the internals and rules.
-   **Growing community**: Used by [Facebook](https://code.facebook.com/posts/879890885467584/improving-css-quality-at-facebook-and-beyond/), [Github](https://github.com/primer/stylelint-config-primer), [Wikimedia](https://github.com/wikimedia/stylelint-config-wikimedia), [GSA](https://github.com/18F/stylelint-rules/), and [WordPress](https://github.com/ntwb/stylelint-config-wordpress/) among others.

## Example output

Example

## Getting started

With stylelint, it's easy to start linting your CSS:

1.  Decide how you want to use stylelint:
    -   [via the stylelint CLI](docs/user-guide/cli.md)
    -   [via a plugin for your build tool](docs/user-guide/complementary-tools.md#build-tool-plugins) (gulp, webpack etc)
    -   [via a plugin for your text editor](docs/user-guide/complementary-tools.md#editor-plugins) (atom, sublime text etc)
    -   [via the stylelint Node API](docs/user-guide/node-api.md)
    -   [via the stylelint PostCSS plugin](docs/user-guide/postcss-plugin.md)
2.  Create your [configuration object](docs/user-guide/configuration.md) by either extending a shared config or crafting your own:
    -   To extend a shared config, we recommend using [`stylelint-config-standard`](https://github.com/stylelint/stylelint-config-standard). It includes over 80 of stylelint's rules with sensible defaults. (You can always override specific rules after extending the config.) We update the config with each new release of stylelint. Alternately, you can [search for](https://www.npmjs.com/browse/keyword/stylelint-config) a community config and [extend](docs/user-guide/configuration.md#extends) that instead.
    -   To craft your own config, first [learn about how rules are named and how they work together](docs/user-guide/about-rules.md), then either:
        -   Start small and only learn about [the rules](docs/user-guide/rules.md) you want to turn on and enforce. *All of the rules are off by default*, and so you can start small, growing your config over time as you have a chance to explore more of the rules.
        -   Or copy-paste [this example configuration](docs/user-guide/example-config.md), which lists all of stylelint's rules and their primary options. Then you can edit the options of each rule to your liking, and remove (or turn off with `null`) the rules that you don't care to enforce.
3.  Lint!

## Guides

You'll find more detailed information on using stylelint and tailoring it to your needs in our guides:

-   [User guide](docs/user-guide.md) - Usage, configuration, FAQ and complementary tools.
-   [Developer guide](docs/developer-guide.md) - Contributing to stylelint and writing your own plugins & formatters.

## Need help?

If you're looking for help or have a support question, then check out our [FAQ](docs/user-guide/faq.md) first. If the answer to your problem isn't there, then go to [stackoverflow](https://stackoverflow.com/questions/tagged/stylelint). stackoverflow is a huge Question and Answer community, and tagging your post there with "stylelint" will catch the stylelint team's attention.

If you think you've found a bug or if you have feature request, then create a [new GitHub issue](https://github.com/stylelint/stylelint/issues/new). Be sure to follow the issue template, answering each question, as this helps us greatly in understanding your problem or request.

Upgrading? Please read our [CHANGELOG](CHANGELOG.md) to learn what changes to expect in the latest version, whether that's new features, bug fixes, renamed rules, or whatever else.

## Help out

There is always a lot of work to do, and already well over 150 rules to maintain. So please help out in any way that you can:

-   Chime in on any open [issue](https://github.com/stylelint/stylelint/issues) or [pull request](https://github.com/stylelint/stylelint/pulls).
-   Create, enhance, and debug rules (see our guide to ["Working on rules"](docs/developer-guide/rules.md)).
-   Improve [documentation](docs/).
-   Add new tests to *absolutely anything*.
-   Work on [improving performance of rules](docs/developer-guide/rules.md#improving-the-performance-of-a-new-or-an-existing-rule).
-   Open new issues about your ideas for making stylelint better, and pull requests to show us how your idea works.
-   Create or contribute to ecosystem tools, like the plugins for [Atom](https://github.com/AtomLinter/linter-stylelint) and [Sublime Text](https://github.com/kungfusheep/SublimeLinter-contrib-stylelint).

[License](https://raw.githubusercontent.com/stylelint/stylelint/master/LICENSE)

</details>




  <details>
    <summary><code>yarn why stylelint</code> output</summary>
    <p><code><ul><li>Has been hoisted to "stylelint"</li><li>This module exists because it's specified in "devDependencies".</li><li>Disk size without dependencies: "3.41MB"</li><li>Disk size with unique dependencies: "12.28MB"</li><li>Disk size with transitive dependencies: "31.57MB"</li><li>Number of shared dependencies: 117
    </li></ul></code></p>
  </details>
  


<h2><a href="https://github.com/stylelint/stylelint-config-standard#readme">stylelint-config-standard</a></h2>
<p>Author: stylelint</p>
<p>Description: Standard shareable config for stylelint</p>
<p>Homepage: <a href="https://github.com/stylelint/stylelint-config-standard#readme">https://github.com/stylelint/stylelint-config-standard#readme</a></p>

<table>
  <thead><tr><th></th><th width="100%"></th></tr></thead>
  <tr><td>Created</td><td>over 1 year ago</td></tr><tr><td>Last Updated</td><td>3 days ago</td></tr><tr><td>License</td><td>MIT</td></tr><tr><td>Maintainers</td><td>3</td></tr><tr><td>Releases</td><td>22</td></tr><tr><td>Keywords</td><td>stylelint, stylelint-config and standard</td></tr>
</table>
This README is too long to show.



  <details>
    <summary><code>yarn why stylelint-config-standard</code> output</summary>
    <p><code><ul><li>Has been hoisted to "stylelint-config-standard"</li><li>This module exists because it's specified in "devDependencies".</li><li>Disk size without dependencies: "48kB"</li><li>Disk size with unique dependencies: "48kB"</li><li>Disk size with transitive dependencies: "48kB"</li><li>Number of shared dependencies: 0
    </li></ul></code></p>
  </details>
  
<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

**Comment by orta** [comment]
screen shot 2017-07-14 at 12 49 13

Alright, this is now good to go. 

This PR adds a command `yarn stylelint-project` which will verify, I'm not running this on CI because we cannot do an auto-fix, you'll get errors in your editor inline, so that's probably enough until it becomes an issue.

**Comment by l2succes** [comment]
Nice! I had stylelint setup in a branch as well but I couldn't figure out the auto-fix part. But yours is more useful because we get error highlighting in the editor.

Let's merge once it's green

**Comment by orta** [comment]
Great - these fails are legit

**Comment by alloy** [comment]
Noice 🌟

File: .stylelintrc [hunk 1]
```diff
@@ -0,0 +1,11 @@
+{
+  "processors": ["stylelint-processor-styled-components"],
+  "extends": "stylelint-config-standard",
+  "syntax": "scss",
+  "rules": {
+    "property-no-unknown": [true, { "ignoreProperties": ["resize-mode"] }],
```
Comment (orta) [inline .stylelintrc]: I can't add comments in this file, so these are RN specific keys

File: .stylelintrc [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+    "block-opening-brace-space-after": null,
+    "block-closing-brace-space-before": null,
```
Comment (orta) [inline .stylelintrc]: These are because prettier will truncate single lined CSS elements into a single LOC

File: .stylelintrc [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+    "declaration-block-no-redundant-longhand-properties": null
```
Comment (orta) [inline .stylelintrc]: RN supports different rules for CSS occasionally, following this linter will get unexpected fails

Comment (orta) [inline .stylelintrc]: The rules are about things like `margin: 20px 10px 30px` which corresponds to `20px 10px 30px 10px` in browsers, but not in RN

**Comment by orta** [comment]
OK, this is good to go

**Pull request merged**